src/measure_theory/lp_space.lean

```diff
@@ -94,4 +94,70 @@ by simp [snorm, hp0_lt]
 
 end zero
 
+@[simp] lemma snorm_neg {f : α → F} : snorm (-f) p μ = snorm f p μ :=
+by simp [snorm]
+
+
+section borel_space
+variable [borel_space E]
+
+lemma mem_ℒp.neg {f : α → E} (hf : mem_ℒp f p μ) : mem_ℒp (-f) p μ :=
+⟨measurable.neg hf.1, by simp [hf.right]⟩
+
+
+variable [topological_space.second_countable_topology E]
+
+lemma mem_ℒp.add {f g : α → E} (hf : mem_ℒp f p μ) (hg : mem_ℒp g p μ) (hp1 : 1 ≤ p) :
+  mem_ℒp (f+g) p μ :=
+begin
+  have hp0_lt : 0 < p, from lt_of_lt_of_le zero_lt_one hp1,
+  have hp0 : 0 ≤ p, from le_of_lt hp0_lt,
+  split,
+  { exact measurable.add hf.1 hg.1, },
+  simp_rw [pi.add_apply, ennreal.coe_rpow_of_nonneg _ hp0],
+  -- step 1: use nnnorm_add_le
+  calc ∫⁻ (a : α), ↑(nnnorm (f a + g a) ^ p) ∂μ ≤ ∫⁻ a, ↑((nnnorm (f a) + nnnorm (g a)) ^ p) ∂ μ :
+  begin
+    refine lintegral_mono_nnreal (λ a, _),
+    exact nnreal.rpow_le_rpow (nnnorm_add_le (f a) (g a)) (le_of_lt hp0_lt)
+  end
+  -- step 2: use convexity of rpow
+  ... ≤ ∫⁻ a, ↑((2:nnreal)^(p-1) * (nnnorm (f a)) ^ p + (2:nnreal)^(p-1) * (nnnorm (g a)) ^ p) ∂ μ :
+  begin
+    refine lintegral_mono_nnreal (λ a, _),
+    have h_zero_lt_half_rpow : (0 : nnreal) < (1 / 2) ^ p,
+    { rw [←nnreal.zero_rpow (ne_of_lt hp0_lt).symm, nnreal.rpow_lt_rpow_iff hp0_lt],
+      simp [zero_lt_one], },
+    have h_rw : (1 / 2) ^ p * (2:nnreal) ^ (p - 1) = 1 / 2,
+    { rw [nnreal.rpow_sub two_ne_zero, nnreal.div_rpow, nnreal.one_rpow, nnreal.rpow_one,
+        ←mul_div_assoc, one_div, inv_mul_cancel],
+      simp [two_ne_zero], },
+    rw [←mul_le_mul_left h_zero_lt_half_rpow, mul_add, ← mul_assoc, ← mul_assoc, h_rw,
+      ←nnreal.mul_rpow, mul_add],
+    refine nnreal.rpow_arith_mean_le_arith_mean2_rpow (1/2 : nnreal) (1/2 : nnreal)
+      (nnnorm (f a)) (nnnorm (g a)) _ hp1,
+    rw [nnreal.div_add_div_same, one_add_one_eq_two, nnreal.div_self two_ne_zero]
+  end
+  -- step 3: use hypotheses hf and hg
+  ... < ⊤ :
+  begin
+    simp_rw [ennreal.coe_add, ennreal.coe_mul, ←ennreal.coe_rpow_of_nonneg _ hp0],
+    rw [lintegral_add, lintegral_const_mul, lintegral_const_mul, ennreal.add_lt_top],
```

Comment:
After this `rw`, you have 5 goals. Each of them should be enclosed between braces.